src/services/ingredients/routes/ingredients.ts

```diff
@@ -5,67 +5,72 @@ import { makeErrorResponse } from '../../../utils/response-error';
 import type { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
 import type { Unit } from './unit';
 
-interface Ingredient {
+export interface Ingredient {
   id: string;
   name: string;
   unit: Unit;
   quantity: number;
   price: number;
 }
 
-interface IngredientsDatabase {
-  [id: string]: Ingredient;
-}
-
 export default async function (fastify: FastifyInstance) {
 
-  const db: IngredientsDatabase = {
-    'abc': {
-      id: 'abc',
-      name: 'Butter',
-      unit: {
-        id: 'def',
-        name: 'kilograms',
-        shortName: 'kg'
-      },
-      quantity: 200,
-      price: 3.50
-    }
-  };
+  fastify.get('/', async (_, reply: FastifyReply) => {
+    try {
+      const ingredients = await fastify.repositories.ingredients.find();
 
-  fastify.get('/', async () => Object.values(db));
+      reply.status(200);
+
+      return ingredients; 
+    } catch (e) {
+      reply.status(400);
+
+      return makeErrorResponse(400, 'There was an error getting the ingredients')
+    }
+  });
 
   fastify.get('/:id', async (request: FastifyRequest<{
     Params: {
       id: string
     }
   }>, reply: FastifyReply) => {
-    const ingredientId = request.params.id;
-    const ingredient = db[ingredientId];
-
-    if (ingredient) {
-      return ingredient;
+    try {
+      const ingredient = await fastify.repositories.ingredients.findOne(request.params.id);
+      
+      if (ingredient) {
+        return ingredient;
+      }
+    } catch (e) {
+      reply.status(400);
+
+      console.log(e)
+
+      return makeErrorResponse(400, `Ingredient with ID: "${request.params.id}" doesn't exist`);
     }
-
-    reply.status(400);
-
-    return makeErrorResponse(400, `Ingredient with ID: "${ingredientId}" doesn't exist`);
   });
 
   fastify.post('/', async (request: FastifyRequest<{
     Body: Ingredient
   }>, reply: FastifyReply) => {
-    const id = uuid();
-    const ingredient = {
-      ...request.body,
-      id
-    };
+    try {
+      const newIngredient = await fastify.repositories.ingredients.create({
+        name: request.body.name,
```

Comment:
Everything here comes from `request.body`. I suggest:

```suggestion
const { name, unit, quantity, price } = request.body;

// --snip--
await fastify.repositories.ingredients.create({
  name,
  unit,
// --snip--
```

Reply:
suggestion applied! Thanks.